app/models/user.rb

```diff
@@ -21,34 +21,39 @@ class User < ApplicationRecord
   has_many :badges, through: :badge_achievements
   has_many :collections, dependent: :destroy
   has_many :comments, dependent: :destroy
-  has_many :email_messages, class_name: "Ahoy::Message"
+  has_many :email_messages, class_name: "Ahoy::Message", dependent: :destroy
   has_many :github_repos, dependent: :destroy
   has_many :identities, dependent: :destroy
   has_many :mentions, dependent: :destroy
   has_many :messages, dependent: :destroy
   has_many :notes, as: :noteable, inverse_of: :noteable
-  has_many :profile_pins, as: :profile, inverse_of: :profile
-  has_many :authored_notes, as: :author, inverse_of: :author, class_name: "Note"
+  has_many :profile_pins, as: :profile, inverse_of: :profile, dependent: :delete_all
```

Comment:
Maybe because it's faster, since `delete_all` directly deletes from the DB. Makes sense since there are no other associations on `ProfilePin` we'd want to trigger here via `:destroy`.

Reply:
Yep, that's for faster deletion. In this case, I `delete_all` will be called anyway because I call it explicitly in `Users::DeleteActivity`, so that shouldn't affect the logic. I don't have a strong opinion if `delete_all` or `destroy` is needed here. As I know, all user deletions should be done via `Users::Delete`, not directly.

Reply:
makes sense! thank you for the explanation